ludwig/benchmarking/profiler_callbacks.py

```diff
@@ -0,0 +1,56 @@
+from typing import Any, Dict
+
+from ludwig.benchmarking.profiler import LudwigProfiler
+from ludwig.callbacks import Callback
+from ludwig.constants import EVALUATION, PREPROCESSING, TRAINING
+
+
+class LudwigProfilerCallback(Callback):
+    """Class that defines the methods necessary to hook into process."""
+
+    def __init__(self, experiment: Dict[str, Any]):
+        self.experiment_name = experiment["experiment_name"]
+        self.use_torch_profiler = experiment["profiler"]["use_torch_profiler"]
+        self.logging_interval = experiment["profiler"]["logging_interval"]
+        self.preprocess_profiler = None
+        self.train_profiler = None
+        self.evaluation_profiler = None
+
+    def on_preprocess_start(self, *args, **kwargs):
+        self.preprocess_profiler = LudwigProfiler(
+            tag=PREPROCESSING,
+            output_dir=self.experiment_name,
+            use_torch_profiler=self.use_torch_profiler,
+            logging_interval=self.logging_interval,
+        )
+        self.preprocess_profiler.__enter__()
+
+    def on_preprocess_end(self, *args, **kwargs):
+        self.preprocess_profiler.__exit__(None, None, None)
+        del self.preprocess_profiler
+
+    def on_train_start(self, *args, **kwargs):
+        self.train_profiler = LudwigProfiler(
+            tag=TRAINING,
+            output_dir=self.experiment_name,
+            use_torch_profiler=self.use_torch_profiler,
+            logging_interval=self.logging_interval,
+        )
+        self.train_profiler.__enter__()
+
+    def on_train_end(self, *args, **kwargs):
+        self.train_profiler.__exit__(None, None, None)
+        del self.train_profiler
+
+    def on_evaluation_start(self):
```

Comment:
probably need to consume args and kwargs here

Reply:
I believe this callback doesn't take any args https://github.com/ludwig-ai/ludwig/blob/2ffd276976f882cab7baa7958e8cb43441b7bdc2/ludwig/callbacks.py#L331